htmlreport/cppcheck-htmlreport

```diff
@@ -423,20 +424,14 @@ if __name__ == '__main__':
     if options.source_dir:
         source_dir = options.source_dir
 
-    # Get the stream that we read cppcheck errors from.
-    input_file = sys.stdin
-    if options.file:
-        if not os.path.exists(options.file):
-            parser.error('cppcheck xml file: %s not found.' % options.file)
-        input_file = io.open(options.file, 'r')
-    else:
-        parser.error('No cppcheck xml file specified. (--file=)')
-
-    # Parse the xml file and produce a simple list of errors.
+    # Parse the xml from all files defined in file argument
+    # or from stdin. If no input is provided, stdin is used
+    # Produce a simple list of errors.
     print('Parsing xml report.')
     try:
         contentHandler = CppCheckHandler()
-        xml_parse(input_file, contentHandler)
+        for fname in options.file or [sys.stdin]:
+            xml_parse(fname, contentHandler)
```

Comment:
Doesn't `xml_parse` expect a file input? Now it just gets a filename?

It was a good idea to write the error: `cppcheck xml file: %s not found.` in case there is a typo in the command line. I would suggest that the script should even terminate when this happens so the user will get a clear indication that something bad happened..

Reply:
xml_parse supports file and filename too as defined in https://docs.python.org/2/library/xml.sax.html:
```
xml.sax.parse(filename_or_stream, handler[, error_handler])
```
If no error_handler provided, then an exception is raised with this content:
```
cppcheck-htmlreport --report-dir=/tmp --file missing_file
Parsing xml report.
Traceback (most recent call last):
  File "/usr/bin/cppcheck-htmlreport", line 429, in <module>
    xml_parse(fname, contentHandler)
  File "/usr/lib/python3.8/xml/sax/__init__.py", line 33, in parse
    parser.parse(source)
  File "/usr/lib/python3.8/xml/sax/expatreader.py", line 105, in parse
    source = saxutils.prepare_input_source(source)
  File "/usr/lib/python3.8/xml/sax/saxutils.py", line 365, in prepare_input_source
    f = urllib.request.urlopen(source.getSystemId())
  File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 509, in open
    req = Request(fullurl, data)
  File "/usr/lib/python3.8/urllib/request.py", line 328, in __init__
    self.full_url = url
  File "/usr/lib/python3.8/urllib/request.py", line 354, in full_url
    self._parse()
  File "/usr/lib/python3.8/urllib/request.py", line 383, in _parse
    raise ValueError("unknown url type: %r" % self.full_url)
ValueError: unknown url type: 'missing_file'
```
Probably ValueError exception could ba caught to provide your error message instead of this unhandled exception.